mne/time_frequency/csd.py

```diff
@@ -155,33 +171,9 @@ def compute_epochs_csd(epochs, mode='multitaper', fmin=0, fmax=np.inf,
 
     # Preparing for computing CSD
     logger.info('Computing cross-spectral density from epochs...')
-    if mode == 'multitaper':
-        # Compute standardized half-bandwidth
-        if mt_bandwidth is not None:
-            half_nbw = float(mt_bandwidth) * n_times / (2 * sfreq)
-        else:
-            half_nbw = 2
-
-        # Compute DPSS windows
-        n_tapers_max = int(2 * half_nbw)
-        window_fun, eigvals = dpss_windows(n_times, half_nbw, n_tapers_max,
-                                           low_bias=mt_low_bias)
-        n_tapers = len(eigvals)
-        logger.info('    using multitaper spectrum estimation with %d DPSS '
-                    'windows' % n_tapers)
-
-        if mt_adaptive and len(eigvals) < 3:
-            warn('Not adaptively combining the spectral estimators due to a '
-                 'low number of tapers.')
-            mt_adaptive = False
-    elif mode == 'fourier':
-        logger.info('    using FFT with a Hanning window to estimate spectra')
-        window_fun = np.hanning(n_times)
-        mt_adaptive = False
-        eigvals = 1.
-        n_tapers = None
-    else:
-        raise ValueError('Mode has an invalid value.')
+    pars = _compute_csd_params(n_times, sfreq, mode, mt_bandwidth, mt_low_bias,
+                               mt_adaptive)
+    window_fun, eigvals, n_tapers, mt_adaptive = pars
```

Comment:
FYI to make this PEP8 but a little bit more compact you can do:
```
window_fun, eigvals, n_tapers, mt_adaptive = _compute_csd_params(
    n_times, sfreq, mode, mt_bandwidth, mt_low_bias, mt_adaptive)
```
...but don't really need to do it here